internal/pkg/gateway/commit/channelnotifier.go

```diff
@@ -13,37 +13,46 @@ import (
 )
 
 type channelLevelNotifier struct {
-	lock          sync.Mutex
 	commitChannel <-chan *ledger.CommitNotification
-	listeners     map[string][]*channelLevelListener
+	lock          sync.Mutex
+	listeners     map[string][]*transactionListener
+	done          <-chan struct{}
 }
 
-func newChannelNotifier(commitChannel <-chan *ledger.CommitNotification) *channelLevelNotifier {
+func newChannelNotifier(done <-chan struct{}, commitChannel <-chan *ledger.CommitNotification) *channelLevelNotifier {
 	notifier := &channelLevelNotifier{
 		commitChannel: commitChannel,
-		listeners:     make(map[string][]*channelLevelListener),
+		listeners:     make(map[string][]*transactionListener),
+		done:          done,
 	}
 	go notifier.run()
 	return notifier
 }
 
 func (notifier *channelLevelNotifier) run() {
 	for {
-		blockCommit, ok := <-notifier.commitChannel
-		if !ok {
-			break
+		select {
+		case blockCommit, ok := <-notifier.commitChannel:
+			if !ok {
+				panic("commit channel closed unexpectedly")
```

Comment:
I'm not sure I understand this. The done channel should be provided for cancelation; if it's closed, it shouldn't bring down the whole peer, should it?

Reply:
If `commitChannel` closes the ledger is dead and transactions can't be processed. I'm told that the design of the API I'm using to obtain `commitChannel` is that it is an error to obtain it more than once. Maybe this can be improved upon. Maybe we can do that in another PR

Reply:
Then I guess I question why have the channel at all. If we're going to terminate the process for on optional feature, why wouldn't the ledger, at the heart of the system, terminate the process?

So, I still don't understand why we'd panic but it can be revisited at a later time.

Reply:
I can change it back to just silently stop if that makes you happier

Reply:
Changed to not panic